andrewckinstler/fitlit-refactor
# Class restructure

#### What's this PR do?
This PR restructures the Sleep and Hydration classes to inherit from the UserParent class, as well as removes redundancies from their respective tests. 
#### Where should the reviewer start?
Start with the UserParent class to get an idea of how the methods were made more dynamic so that any of the classes could use them. 
#### How should this be manually tested?
I would pull the branch and run the tests on your machine. I would love some feedback on whether you think additional tests should be added to UserParent for these two classes.
#### Any background context you want to provide?
Just a note that currently the tests are running off of the hydration class. I don't foresee issues with the other classes, but I have not explicitly tested for them. 

@jpweist @brittanystoroz @khalidwilliams
